lib/block_view_dao_coin_limit_order.go

```diff
@@ -366,183 +377,79 @@ func (bav *UtxoView) _connectDAOCoinLimitOrder(
-				// amount purchased by the transactor.
-				filledOrders = append(filledOrders, &FilledDAOCoinLimitOrder{
-					TransactorPKID:                 transactorPKIDEntry.PKID,
-					BuyingDAOCoinCreatorPKID:       buyCoinPKIDEntry.PKID,
-					SellingDAOCoinCreatorPKID:      sellCoinPKIDEntry.PKID,
-					BuyingDAOCoinQuantityPurchased: buyCoinBaseUnitsBought,
-					BuyingDAOCoinQuantityRequested: transactorOriginalQuantityToBuyInBaseUnits,
-					SellingDAOCoinQuantitySold:     sellCoinBaseUnitsSold,
-				})
-
-				// If we're here it means we fully covered the transactor's order
-				// so we're done
+			if updatedTransactorOrderQuantityToFill.IsZero() {
+				// Transactor's order was fulfilled.
+				transactorOrder.QuantityToFillInBaseUnits = uint256.NewInt()
 				orderFilled = true
+				transactorOrderFilledOrder.IsFulfilled = true
 			} else {
-				// Since the transactor's order's quantity is greater than or equal to the matching
-				// order's quantity, we transfer the matching order's quantity. This case is a lot
-				// easier because we don't need to do much interpolation.
-				//
-				// Remember that the transactor order is doing (sell A, buy B) while the matching order
-				// is doing (sell B, buy A). That means that the number of B units being BOUGHT is
-				// actually the number of B units that the matching order is SELLING.
-				buyCoinBaseUnitsBought = matchingOrderTotalBuyCoinToSell
-
-				transactorOriginalQuanityToBuyInBaseUnits := transactorOrder.QuantityToBuyInBaseUnits
-				// Update transactor order's quantity.
-				transactorOrder.QuantityToBuyInBaseUnits, err = SafeUint256().Sub(
-					transactorOriginalQuanityToBuyInBaseUnits, matchingOrderTotalBuyCoinToSell)
-				if err != nil {
-					return 0, 0, nil, errors.Wrapf(err, "Error updating order quantity: ")
-				}
+				// Transactor's order is incomplete. Note we don't store the
+				// transactor order in the db until we have finished looping
+				// through all matching orders.
+				transactorOrder.QuantityToFillInBaseUnits = updatedTransactorOrderQuantityToFill
+				transactorOrderFilledOrder.IsFulfilled = false
+			}
 
-				// Save the quantity of the matching order. This is the number of coins
-				// the transactor is selling.
-				sellCoinBaseUnitsSold = matchingOrder.QuantityToBuyInBaseUnits
+			filledOrders = append(filledOrders, transactorOrderFilledOrder)
 
-				// Mark matching order for deletion.
+			// Update quantity for matching order.
+			matchingOrderFilledOrder := &FilledDAOCoinLimitOrder{
+				TransactorPKID:                            matchingOrder.TransactorPKID,
+				BuyingDAOCoinCreatorPKID:                  matchingOrder.BuyingDAOCoinCreatorPKID,
+				SellingDAOCoinCreatorPKID:                 matchingOrder.SellingDAOCoinCreatorPKID,
+				BuyingDAOCoinQuantityInBaseUnitsPurchased: daoCoinBaseUnitsSoldByTransactor,
+				SellingDAOCoinQuantityInBaseUnitsSold:     daoCoinBaseUnitsBoughtByTransactor,
```

Comment:
I think I incorrectly named these two struct attributes, should these really be `BuyingCoinQuantityInBaseUnitsPurchased` and `SellingCoinQuantityInBaseUnitsPurchased`?

Reply:
I think this is right. It shows for the given order how much DAO coin base units they bought and sold. It's inverted depending on whether you're looking at the transactor or the matching order. I.e. the # DAO coins bought by the transactor == # DAO coins sold by the matching order and vice versa. Which is why you see them flipped here. We're looking at the matching order.

Reply:
These can't be amounts of DESO?

Reply:
You're right. Updating now.